cmd/zpool/compatibility.d/grub2-2.06

```diff
@@ -0,0 +1,23 @@
+# Features which are supported by GRUB2 versions prior to v2.12.
```

Comment:
Having this compatibility file named `grub2-2.06` yet reference v2.12 as the cut off is a bit confusing.  Is this because most/all distributions jumped from v2.06 to v2.12 and there effectively aren't versions in between out in the wild?

Reply:
I don't see a confusion in specifying the oldest version supported by respective file.  I think when somebody chooses among different versions, he should be conservative and select biggest available version smaller or equal to his requirements.  This file supports versions [2.06, 2.12).  The confusion here is that version supporting 2.12 and up is called just "grub2".

Reply:
Yes, that's unfortunate.  Rather than live with this confusion forever what if we renamed the "grub2" file to "grub2-2.12" for the 2.3 release.  Then put in place a "grub2" -> "grub2-2.12" symlink.  We could also backport this change to the 2.2 releases.  Once in place we could consider retiring the "grub2" file in a future release.  Users would then need to specifically set "compatibility=grub2-2.12" if they were using the "grub2" config.

Reply:
Yea, something like that could probably be done.  I just don't remember what is the reaction of zpool on absent compatibility file.  We could look on how to make it less annoying for user and still notify that intervention is required.

Reply:
That sounds good to me.  We don't even need to be in a rush to remove the "grub2" symlink.  Although, I agree it'd be good to have some notification/warning that encourage users to update their `compatibility` property.  We'd also want to discourage new pools from being created using the "grub2" symlink.

Reply:
I have renamed the `grub2` compatibility file to `grub2-2.12` since it contains the features set that is compatible with grub2 version 2.12 onwards.

A new symlink named `grub2` is created that points to `grub2-2.12` file.